db/migrate/20210120100145_create_jobs.rb

```diff
@@ -0,0 +1,15 @@
+class CreateJobs < ActiveRecord::Migration[5.2]
+  def change
+    create_table :jobs do |t|
```

Comment:
we discussed with Camille that we will need to have a separate relation table, smth like:
job_id
reference_type
reference_id

This will help us to solve:
1. When one job has two restrictions (for maintenance and for whitelist during the maintenance)
2. Keep flexibility here to have the ability to create Job not only for Restrictions

Reply:
I create a new intermediate table call "jobbings" as picture below.
![Barong Jobs](https://user-images.githubusercontent.com/10076938/105445860-8c684380-5ca3-11eb-973d-a253c006150e.png)

I can now create Job, Restriction and relationships like this
```ruby

# Get maintenance restriction
maintenance = Restriction.find_or_create_by(category: 'maintenance', scope: 'all', value: 'all', code: 471, state: 'disabled')

# Create new whitelist restriction for admin
whitelist = Restriction.find_or_create_by(category: 'whitelist', scope: 'ip', value: '164.116.47.255', code: 433, state: 'disabled')

# Create new maintenance job
job = Job.create!(type: 'maintenance', description: 'monthly maintenance', state: 'pending', start_at: "2020-01-31T00:00:00", finish_at: '2020-01-31T03:00:00')

# Create relationship
Jobbing.create!(job: job, reference: maintenance)
Jobbing.create!(job: job, reference: whitelist)

# Get all restrictions by job
restrictions = job.restrictions # or job.jobbings
````



If other models need to create new Job it can use polymorphic relationship feature in Rails.